fe1-web/src/core/objects/Hash.ts

```diff
@@ -78,4 +80,37 @@ export class Hash extends Base64UrlData {
     const byteArray = new TextEncoder().encode(str);
     return byteArray.length;
   }
+
+  /**
+   * Returns *some* string representation of this object.
+   * If you need access to the unterlying data type use .valueOf() and
+   * if you want to serialize an instance use .toState() instead
+   */
+  public toString(): string {
+    return super.toString();
+  }
+
+  /**
+   * Returns the primitive value used for representing the Hash,
+   * a string
+   * If you want to serialize an instance use .toState() instead
+   */
+  public valueOf(): string {
+    return super.valueOf();
+  }
+
+  /**
+   * Returns the serialized version of the hash that can for instance be stored
+   * in redux stores
+   */
+  public toState(): HashState {
+    return super.valueOf();
+  }
```

Comment:
aren't you redefining / hiding the inherited methods here, whilst providing an identical implementation ?

Reply:
Mostly in order to add the comments that describe the purpose of the different functions. But to make this more explicit and hopefully avoid misuse I could also see `toString` to be changed for most of these to something like ```
```
return `Hash(${super.toString()})`;
```

or as you commented for timstamps the ISO date

Reply:
I like your suggestion, but I'm concerned that there are still cases where we use `toString` instead of `valueOf` 🤔 
Don't you think this is typical one of those changes that risks breaking stuff everywhere? 😅

Reply:
Yeah... Usually I would like argue it would only break in places that would require fixing since it will just fail at a later point but given the current state I think I agree that we might not want to do that right now 😅😅